### ISSUE TITLE
Graal Chrome Inspector throwing java.lang.NullPointerException: Cannot read field "engineIndex" because "language" is null

### ISSUE THREAD

**Issue by ascertsimonm** [oracle/graal#5308]
When using the Graal VM JS Engine running as a replacement for Nashorn under Java 17 and using the Chrome Inspector to debug JavaScript the following exception is thrown

`java.lang.NullPointerException: Cannot read field "engineIndex" because "language" is null`

(see end for full, long stack trace). I initially found the issue using Graal version 22.3.0 going back using older versions, 22.1.0.1 does not appear to have the same error.

The error occurs when the JavaScript has reference to a Java Object defined in the calling Java code, for example the TestInnerClass object in my example.  Standard Java objects, such as java.lang.String, on their own do not cause the exception.

This can be reproduced using same code as I created for this issue [here](https://github.com/oracle/graal/issues/3822).  

Debugging the Graal code it seems to be a problem around the recognition/registration of the 'host' language although I couldn't pin down exactly why.

Cheers
Simon

This is the full stack shown in the Chrome console window.

```
java.lang.NullPointerException: Cannot read field "engineIndex" because "language" is null
	at com.oracle.truffle.polyglot.PolyglotContextImpl.getContext(PolyglotContextImpl.java:695)
	at com.oracle.truffle.polyglot.PolyglotContextImpl.getContextInitialized(PolyglotContextImpl.java:703)
	at com.oracle.truffle.polyglot.EngineAccessor$EngineImpl.getLanguageView(EngineAccessor.java:226)
	at com.oracle.truffle.api.instrumentation.TruffleInstrument$Env.getLanguageView(TruffleInstrument.java:916)
	at com.oracle.truffle.api.debug.DebugValue.getLanguageView(DebugValue.java:1148)
	at com.oracle.truffle.api.debug.DebugValue.getMetaObject(DebugValue.java:1177)
	at com.oracle.truffle.tools.chromeinspector.types.RemoteObject.getMetaObject(RemoteObject.java:345)
	at com.oracle.truffle.tools.chromeinspector.types.TypeInfo.fromValue(TypeInfo.java:105)
	at com.oracle.truffle.tools.chromeinspector.types.RemoteObject.initFromValue(RemoteObject.java:129)
	at com.oracle.truffle.tools.chromeinspector.types.RemoteObject.<init>(RemoteObject.java:113)
	at com.oracle.truffle.tools.chromeinspector.types.RemoteObject.<init>(RemoteObject.java:98)
	at com.oracle.truffle.tools.chromeinspector.InspectorRuntime.createPropertyJSON(InspectorRuntime.java:932)
	at com.oracle.truffle.tools.chromeinspector.InspectorRuntime.createPropertyJSON(InspectorRuntime.java:922)
	at com.oracle.truffle.tools.chromeinspector.InspectorRuntime.createPropertyJSON(InspectorRuntime.java:918)
	at com.oracle.truffle.tools.chromeinspector.InspectorRuntime.putResultProperties(InspectorRuntime.java:450)
	at com.oracle.truffle.tools.chromeinspector.InspectorRuntime$4.executeCommand(InspectorRuntime.java:369)
	at com.oracle.truffle.tools.chromeinspector.InspectorRuntime$4.executeCommand(InspectorRuntime.java:358)
	at com.oracle.truffle.tools.chromeinspector.InspectorExecutionContext$1.run(InspectorExecutionContext.java:251)
	at com.oracle.truffle.tools.chromeinspector.InspectorDebugger$SuspendedCallbackImpl.onSuspend(InspectorDebugger.java:1171)
	at com.oracle.truffle.api.debug.DebuggerSession.doSuspend(DebuggerSession.java:1308)
	at com.oracle.truffle.api.debug.DebuggerSession.processBreakpointsAndStep(DebuggerSession.java:1276)
	at com.oracle.truffle.api.debug.DebuggerSession.notifyCallback(DebuggerSession.java:1082)
	at com.oracle.truffle.api.debug.DebuggerSession$SteppingNode.doStepBefore(DebuggerSession.java:1566)
	at com.oracle.truffle.api.debug.DebuggerSession$SteppingNode.onEnter(DebuggerSession.java:1532)
	at com.oracle.truffle.api.instrumentation.ProbeNode$EventProviderChainNode.innerOnEnter(ProbeNode.java:1382)
	at com.oracle.truffle.api.instrumentation.ProbeNode$EventChainNode.onEnter(ProbeNode.java:913)
	at com.oracle.truffle.api.instrumentation.ProbeNode.onEnter(ProbeNode.java:215)
	at com.oracle.truffle.js.nodes.JavaScriptNodeWrapper.executeVoid(JavaScriptNodeWrapper.java:239)
	at com.oracle.truffle.js.nodes.control.AbstractBlockNode.executeVoid(AbstractBlockNode.java:80)
	at com.oracle.truffle.js.nodes.control.AbstractBlockNode.executeVoid(AbstractBlockNode.java:55)
	at com.oracle.truffle.api.impl.DefaultBlockNode.executeGeneric(DefaultBlockNode.java:63)
	at com.oracle.truffle.js.nodes.control.AbstractBlockNode.execute(AbstractBlockNode.java:75)
	at com.oracle.truffle.js.nodes.binary.DualNode.execute(DualNode.java:119)
	at com.oracle.truffle.js.nodes.function.FunctionBodyNode.execute(FunctionBodyNode.java:73)
	at com.oracle.truffle.js.nodes.instrumentation.DeclareTagProvider$MaterializedFunctionBodyNode.execute(DeclareTagProvider.java:166)
	at com.oracle.truffle.js.nodes.JavaScriptNodeWrapper.execute(JavaScriptNodeWrapper.java:45)
	at com.oracle.truffle.js.nodes.function.FunctionRootNode.executeInRealm(FunctionRootNode.java:149)
	at com.oracle.truffle.js.runtime.JavaScriptRealmBoundaryRootNode.execute(JavaScriptRealmBoundaryRootNode.java:88)
	at com.oracle.truffle.api.impl.DefaultCallTarget.callDirectOrIndirect(DefaultCallTarget.java:85)
	at com.oracle.truffle.api.impl.DefaultDirectCallNode.call(DefaultDirectCallNode.java:59)
	at com.oracle.truffle.js.nodes.function.JSFunctionCallNode$UnboundJSFunctionCacheNode.executeCall(JSFunctionCallNode.java:1314)
	at com.oracle.truffle.js.nodes.function.JSFunctionCallNode.executeAndSpecialize(JSFunctionCallNode.java:306)
	at com.oracle.truffle.js.nodes.function.JSFunctionCallNode.executeCall(JSFunctionCallNode.java:251)
	at com.oracle.truffle.js.nodes.interop.JSInteropExecuteNode.doDefault(JSInteropExecuteNode.java:68)
	at com.oracle.truffle.js.nodes.interop.JSInteropExecuteNodeGen.executeAndSpecialize(JSInteropExecuteNodeGen.java:62)
	at com.oracle.truffle.js.nodes.interop.JSInteropExecuteNodeGen.execute(JSInteropExecuteNodeGen.java:44)
	at com.oracle.truffle.js.runtime.interop.InteropBoundFunction.execute(InteropBoundFunction.java:111)
	at com.oracle.truffle.js.runtime.interop.InteropBoundFunctionGen$InteropLibraryExports$Cached.executeNode_AndSpecialize(InteropBoundFunctionGen.java:189)
	at com.oracle.truffle.js.runtime.interop.InteropBoundFunctionGen$InteropLibraryExports$Cached.execute(InteropBoundFunctionGen.java:172)
	at com.oracle.truffle.api.interop.InteropLibraryGen$Delegate.execute(InteropLibraryGen.java:3886)
	at com.oracle.truffle.api.interop.InteropLibraryGen$CachedDispatch.execute(InteropLibraryGen.java:7765)
	at com.oracle.truffle.polyglot.PolyglotValueDispatch$InteropValue$AbstractExecuteNode.executeShared(PolyglotValueDispatch.java:4268)
	at com.oracle.truffle.polyglot.PolyglotValueDispatch$InteropValue$ExecuteNoArgsNode.executeImpl(PolyglotValueDispatch.java:4375)
	at com.oracle.truffle.polyglot.HostToGuestRootNode.execute(HostToGuestRootNode.java:124)
	at com.oracle.truffle.api.impl.DefaultCallTarget.callDirectOrIndirect(DefaultCallTarget.java:85)
	at com.oracle.truffle.api.impl.DefaultCallTarget.call(DefaultCallTarget.java:102)
	at com.oracle.truffle.api.impl.DefaultRuntimeAccessor$DefaultRuntimeSupport.callProfiled(DefaultRuntimeAccessor.java:182)
	at com.oracle.truffle.polyglot.PolyglotValueDispatch$InteropValue.execute(PolyglotValueDispatch.java:2401)
	at org.graalvm.polyglot.Value.execute(Value.java:878)
	at org.example.GraalTest.main(GraalTest.java:54)
```

**Comment by oubidar-Abderrahim** [comment]
Hi, Thank you for reporting this issue. it is tracked internally on GR 42207

**Comment by iannakin** [comment]
Hi, 
I also face the same issue using Java 17 (with Kotlin) and GraalVM 22.3.0, with the exact same stacktrace.
DebugValue.getMetaObject is invoked with language "host" as parameter, which seem to not be found in the context languages array.
Some updates on this issue would be much appreciated, thanks !
Cheers,

**Comment by oubidar-Abderrahim** [comment]
Hi @iannakin, 
This issue should be fixed in GraalVM 22.3.1, please give it a try, and if you still face the same error, reopen the ticket. 
Thank you.

**Comment by iannakin** [comment]
Yes indeed, thanks !
